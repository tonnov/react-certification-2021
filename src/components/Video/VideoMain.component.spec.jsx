import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { VideoMain } from './VideoMain.component';
import infoVideo from '../../mock/single-video.json';
import GlobalProvider from '../../providers/Global';

describe('VideoMain component', () => {
  const embedId = 'T_j60n1zgu0';
  const [selVideo] = infoVideo.items;

  beforeEach(() => {
    render(
      <GlobalProvider>
        <VideoMain embedId={embedId} video={selVideo} />
      </GlobalProvider>
    );
  });

  it('should match snapshot', () => {
    const snap = renderer.create(
      <GlobalProvider>
        <VideoMain embedId={embedId} video={selVideo} />
      </GlobalProvider>
    );
    expect(snap.toJSON()).toMatchSnapshot();
  });

  it('should render an iframe element', () => {
    const iframe = screen.getByTestId('iframe');
    expect(iframe).toBeInTheDocument();
  });

  // const component = render(<VideoMain embedId={embedId} video={selVideo} />);

  // component.debug();
});
