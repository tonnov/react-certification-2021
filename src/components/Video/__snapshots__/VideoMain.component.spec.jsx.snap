// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoMain component should match snapshot 1`] = `
Array [
  <div
    className="sc-iCoGMd fyXPXE"
  >
    <iframe
      allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
      allowFullScreen={true}
      data-testid="iframe"
      frameborser="0"
      height="480"
      src="https://www.youtube.com/embed/T_j60n1zgu0"
      title="Embedded Youtube Video"
      width="854"
    />
  </div>,
  <div
    className="sc-bdnxRM bFAezh"
  >
    <h3
      className="sc-gtsrHT idTysY"
    >
      Curso REACT JS ⚛️ - Aprende desde CERO 📈 Componentes, State, JSX (Tutorial Desde Cero en Español)
    </h3>
    <span
      className="sc-dlnjwi qyasb"
    >
      <button
        aria-describedby={null}
        aria-label="add to favorites"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Add to Favorites"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "color": "#616161",
                "fontSize": 27,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </span>
  </div>,
  <div
    className="sc-hKFxyN cjIctm"
  >
    <div
      className="sc-eCApnc gPnuTA"
    >
      Publicado: 4/11/2020
    </div>
    <div
      className="sc-jSFjdj ceWepK"
    >
      99,052 Visualizaciones
    </div>
  </div>,
  <p
    className="sc-gKAaRy dnGvbU"
  >
    Te enseño React desde cero. 👨‍🏫 ¿Qué es React? ¿Por qué deberías aprenderlo? ¿Por qué lo necesitamos? ¿Qué es JSX 🤔? Props y State. Eventos. Renderizado condicional y useState. 🔥
  </p>,
]
`;
